ext/page_log/palm/palm_kv.c

```diff
@@ -112,13 +112,22 @@ swap_page_key(const PAGE_KEY *src, PAGE_KEY *dest)
  */
 #define RESULT_MATCH(result_key, _table_id, _page_id, _now)                          \
     ((result_key)->table_id == (_table_id) && (result_key)->page_id == (_page_id) && \
-      _now > (result_key)->timestamp_materialized_us)
+      (_now) > (result_key)->timestamp_materialized_us)
+
+/*
+ * True if the results match the table, and if the page's version is LTE the given checkpoint.
+ */
+#define RESULT_MATCH_EXT(result_key, _table_id, _page_id, _lsn, _checkpoint_id, _now) \
+    (RESULT_MATCH(result_key, _table_id, _page_id, _now) &&                           \
+      ((_checkpoint_id) == 0 || (result_key)->checkpoint_id <= (_checkpoint_id)) &&   \
+      ((_lsn) == 0 || (result_key)->lsn <= (_lsn)) &&                                 \
```

Comment:
Nit: since `RESULT_MATCH` is not used anywhere else, I would get rid of it, and change the name of `RESULT_MATCH_EXT` to `RESULT_MATCH`.